managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryImpl.java

```diff
@@ -44,6 +44,8 @@ protected EntryImpl newObject(Handle<EntryImpl> handle) {
     private long entryId;
     ByteBuf data;
 
+    private Runnable onDeallocate;
```

Comment:
This is not volatile, but it seems that `deallocate()` could be called in other threads?

Reply:
this is the pattern we use for every recyclable object, the same comment may apply to all the other fields.